src/main/java/com/waytta/clientinterface/LocalClient.java

```diff
@@ -1,13 +1,60 @@
 package com.waytta.clientinterface;
 
+import hudson.Extension;
+import org.kohsuke.stapler.DataBoundConstructor;
+import org.kohsuke.stapler.QueryParameter;
+import hudson.util.FormValidation;
+import org.jenkinsci.Symbol;
+
+import com.waytta.Utils;
+
 public class LocalClient extends BasicClient {
+    public static final int DEFAULT_JOB_POLL_TIME = 10;
+    private Integer jobPollTime = DEFAULT_JOB_POLL_TIME;
+    private Boolean blockbuild = false;
+    private String target;
+    private String targetType;
 
-    public LocalClient(String credentialsId, String target, String targettype, String function, Boolean blockbuild, Integer jobPollTime) {
-        super(credentialsId, target, targettype, function);
+    @DataBoundConstructor
+    public LocalClient(Boolean blockbuild, Integer jobPollTime, String target, String targetType) {
+        this.blockbuild = blockbuild;
+        this.jobPollTime = jobPollTime;
+        this.target = target;
+        this.targetType = targetType;
+    }
+    
+    public Integer getJobPollTime() {
+        return jobPollTime;
+    }
+    
+    public Boolean getBlockbuild() {
+        return blockbuild; 
+    }
+    
+    public String getTarget() {
+        return target;
+    }
 
-        setBlockBuild(blockbuild);
-        setJobPollTime(jobPollTime);
-        setTarget(target);
-        setTargetType(targettype);
+    public String getTargetType() {
+        return targetType;
+    }
+    
+    @Symbol("local")
+    public static final class DescriptorImpl extends BasicClientDescriptor {
+        private DescriptorImpl(Class<? extends BasicClient> clazz) {
+            super(clazz);
+        }
+        
+        @Override
+        public String getDisplayName() {
+        	return "local";
```

Comment:
Conventionally should be capitalized, unless this is intended to match a code word in some external system.